content/ui-testing-handbook/react/en/workflow.md

```diff
@@ -0,0 +1,172 @@
+
+#### PR check
+
+To catch regressions you need to run on all your components. You can do that by importing stories into a Jest and then running an accessibility audit using [jest-axe](https://github.com/twilio-labs/paste/blob/cd0ddad508e41cb9982a693a5160f1b7866f4e2a/packages/paste-core/components/checkbox/__tests__/checkboxdisclaimer.test.tsx#L40). All violations will be reported back to the PR page.
+
+![](/ui-testing-handbook/ci-a11y.png)
+
+### Interaction tests
+
+The user can delete a task by clicking on the _trash can_ button, we’ll need to add in a test to verify that behaviour.
+
+![](/ui-testing-handbook/manual-interaction.gif)
+
+#### During development
+
+During development, manually verify the interaction using the InboxScreen stories. If it’s working as expected, you can move on to adding in an interaction test using Jest and Testing Library.
+
+```diff:title=src/InboxScreen.test.js
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import {
+  render,
+  waitFor,
+  cleanup,
+  within,
+  fireEvent,
+} from '@testing-library/react';
+import { axe, toHaveNoViolations } from 'jest-axe';
+import { composeStories } from '@storybook/testing-react';
+import { getWorker } from 'msw-storybook-addon';
+import * as stories from './InboxScreen.stories';
+
+expect.extend(toHaveNoViolations);
+
+describe('InboxScreen', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  afterAll(() => getWorker().close());
+
+  const { Default } = composeStories(stories);
+
+  it('should pin a task', async () => { ... });
+  it('should archive a task', async () => { ... });
+  it('should edit a task', async () => { ... });
+  it('Should have no accessibility violations', async () => { ... });
+
++ it('should delete a task', async () => {
++   const { queryByText, getByRole, getAllByRole } = render(<Default />);
++
++   await waitFor(() => {
++     expect(queryByText('You have no tasks')).not.toBeInTheDocument();
++   });
++
++   const getTask = () => getByRole('listitem', { name: 'Export logo' });
++
++   const deleteButton = within(getTask()).getByRole('button', {
++     name: 'delete',
++   });
++
++   fireEvent.click(deleteButton);
++
++   expect(getAllByRole('listitem').length).toBe(5);
++ });
+});
+
+```
+
+Run `yarn test` to confirm that all tests are passing. Notice how Jest runs in watch mode and only executes tests related to files that changed.
+
+![](/ui-testing-handbook/jest.png)
+
+#### PR check
+
+Github Actions will run Jest when the pull request is created and report status via PR checks.
+
+![](/ui-testing-handbook/jest-ci.png)
+
+## User flow tests
+
+Lastly, you'll need to run E2E tests to ensure that all your critical user flows are working as expected.
```

Comment:
I expected this section to mention that the pragmatic, hybrid approach you're taking to E2E testing means that we don't need a new integration test for this scenario _and_ the PR check will ensure you didn't break existing critical user flows.

Reply:
Not sure I follow. The delete functionality didn't exist before so, previous tests didn't account for it. 

When you say integration tests, do you mean interaction test?

Reply:
> The delete functionality didn't exist before so, previous tests didn't account for it.

Right. They don't specifically account for it, but, because of the approach you took when defining your E2E tests, they also didn't break when you added those delete buttons. So you can confirm your critical user flows without having to touch the tests themselves.

My suggestion is to make that benefit explicit in the prose, but it's just a suggestion. It's fine as-is.

> When you say integration tests, do you mean interaction test?

Neither. I should've said E2E. Sorry for the confusion.

Reply:
Ok. But, that's exactly what I say, no? We don't touch the e2e tests simply run them on the CI.

Reply:
Or are you suggesting that I call out the fact that we didn't touch the log in functionality so it should work the same as before and the e2e tests confirm that?

Reply:
Yes, precisely.